From 05de992ec439de0e93050105570dcfe17b0a7d90 Mon Sep 17 00:00:00 2001
From: Adnan Begovic <adnan@cyngn.com>
Date: Thu, 2 Jun 2016 15:29:16 -0700
Subject: [PATCH 02/18] SystemUI: Add support for persistent usb drive
 notification.

  This option can be enabled via config_persistUsbDriveNotification

Change-Id: I331a759aeb914e46e3ef4842896d3465d894707a
TICKET: PAELLA-225
---
 core/res/res/values/cm_symbols.xml                                     | 2 ++
 core/res/res/values/config.xml                                         | 3 +++
 .../SystemUI/src/com/android/systemui/usb/StorageNotification.java     | 2 ++
 3 files changed, 7 insertions(+)

diff --git a/core/res/res/values/cm_symbols.xml b/core/res/res/values/cm_symbols.xml
index ed68cb2..17007c2 100644
--- a/core/res/res/values/cm_symbols.xml
+++ b/core/res/res/values/cm_symbols.xml
@@ -20,4 +20,6 @@
     <java-symbol type="array" name="config_disabledComponents" />
     <java-symbol type="array" name="config_forceEnabledComponents" />
 
+    <!-- Usb drive persistent notification -->
+    <java-symbol type="bool" name="config_persistUsbDriveNotification" />
 </resources>
diff --git a/core/res/res/values/config.xml b/core/res/res/values/config.xml
index 9ba745c..625c467 100644
--- a/core/res/res/values/config.xml
+++ b/core/res/res/values/config.xml
@@ -2924,4 +2924,7 @@
     <!-- The list of components which should be forced to be enabled. -->
     <string-array name="config_forceEnabledComponents" translatable="false">
     </string-array>
+
+    <!-- Whether to persist the notification for when a usb drive device is plugged in -->
+    <bool name="config_persistUsbDriveNotification">false</bool>
 </resources>
diff --git a/packages/SystemUI/src/com/android/systemui/usb/StorageNotification.java b/packages/SystemUI/src/com/android/systemui/usb/StorageNotification.java
index e351297..826b79f 100644
--- a/packages/SystemUI/src/com/android/systemui/usb/StorageNotification.java
+++ b/packages/SystemUI/src/com/android/systemui/usb/StorageNotification.java
@@ -393,6 +393,8 @@ public class StorageNotification extends SystemUI {
                             mContext.getString(R.string.ext_media_unmount_action),
                             buildUnmountPendingIntent(vol)))
                     .setContentIntent(browseIntent)
+                    .setOngoing(mContext.getResources().getBoolean(
+                            R.bool.config_persistUsbDriveNotification))
                     .setCategory(Notification.CATEGORY_SYSTEM);
             // Non-adoptable disks can't be snoozed.
             if (disk.isAdoptable()) {
-- 
2.7.4

